solidity/contracts/bank/Bank.sol

```diff
@@ -254,6 +257,31 @@ contract Bank is Ownable {
         _increaseBalance(recipient, amount);
     }
 
+    /// @notice Increases the given smart contract `vault`'s balance and
+    ///         notifies the `vault` contract. Called by the Bridge after
+    ///         the deposits routed by depositors to that `vault` have been
+    ///         swept by the Bridge. This way, the depositor does not have to
+    ///         issue a separate  transaction to the `vault` contract.
+    ///         Can be called only by the Bridge.
+    /// @dev The `vault` must implement `IVault` intrface.
+    /// @param vault Address of `IVault` recipient contract
+    /// @param depositors Addresses of depositors whose deposits have been swept
+    /// @param depositedAmounts Amounts deposited by individual depositors and
+    ///        swept. The `vault`'s balance in the Bank will be increased by the
+    ///        sum of all elements in this array.
+    function increaseBalanceAndCall(
```

Comment:
Non-blocking but worth noting such a signature of `increaseBalanceAndCall` will require grouping deposits by `vault` during the `sweep` transaction which can have a significant gas cost overhead. I'm wondering whether we should add an auxiliary mapping in the Bridge which will address that at `revealDeposit` stage.

Reply:
Right. I am experimenting with the code from #90 to see if we can store the hash of deposit info instead of the entire deposit info. This should let us reduce the gas expenditure by ~25% for most cases. Then, we'd need to pass deposit info as calldata to `sweep` and this is where we can group them together by vault.